core/Pos/Binary/Core/Coin.hs

```diff
@@ -115,20 +120,25 @@ hdrToParam h
     | isClear h 4 = (3, fromIntegral (h .&. 0x0f))
     | otherwise   = (4, fromIntegral (h .&. 0x0f))
 
-decodeVarint :: Peek Word64
-decodeVarint = do
-    (nbBytes, acc) <- hdrToParam <$> getWord8
-    conts <- replicateM nbBytes getWord8
-    let val = orAndShift acc conts
-    when (expectedContBytes val /= length conts) $ fail "not canonical encoding"
-    return val
+decodeMegaMicros :: [Word8] -> Decoder s (Word64, [Word8])
```

Comment:
I'm definitely 👍  to use standard CBOR encoding whenever we can, but I was keen to get an opinion from you guys first.  Older revision of this branch had `Bi Coin` defined simply as:

``` haskell
instance Bi T.Coin where
  encode = encode . T.getCoin
  decode = T.mkCoin <$> decode
```

In a nutshell, as `Coin` is just a wrapper for `Word64`, I would be tempted to simply treat `Coin` as just another newtype and simple encode/decode the inner `Word64`. We can worry about size of the serialised value at a later stage, methinks (if that turns out to even be an issue). @neongreen , @dcoutts What do you guys reckon?